crates/nu-command/src/platform/dir_info.rs

```diff
@@ -106,13 +106,8 @@ impl DirInfo {
         match std::fs::read_dir(&s.path) {
             Ok(d) => {
                 for f in d {
-                    match ctrl_c {
-                        Some(ref cc) => {
-                            if cc.load(Ordering::SeqCst) {
-                                break;
-                            }
-                        }
-                        None => continue,
+                    if nu_utils::ctrl_c::was_pressed(&ctrl_c) {
+                        break;
```

Comment:
The behavior is changed if ctrl_c is None, maybe we can't replace with `was_pressed` function

Reply:
Good catch! You're right that this changes the behaviour. I think it's OK though. 

I think the old `None => continue` behaviour was a bug; `ctrl_c` should not change the behavior of a function.

I checked the callers of `DirInfo::new()` just now, and `ctrl_c` will always be `Some`.